src/worker.js

```diff
@@ -139,6 +139,14 @@ self.onmessage = (e) => {
       Module['dynamicLibraries'] = e.data.dynamicLibraries;
 #endif
 
+      // Use `const` here to ensure that the variable is scoped only to
+      // that iteration, allowing safe reference from a closure.
+      for (const handler of e.data.handlers) {
+        Module[handler] = function() {
```

Comment:
How about only setting this if `handler` doesn't already exist as property of the module?  If the handler is already defined on the module there should be no need to proxy it right?

Reply:
Those handlers (`onExit`, `onAbort`, `print` and `printErr`) are only proxied if it's present in `INCOMING_MODULE_JS_API` and if it's explicitly set on the module.  The latter is controlled by this:
https://github.com/emscripten-core/emscripten/blob/096a5de8e8267ea2ed04d22fc25fa36c2e5cadf0/src/library_pthread.js#L346-L348

So, in most cases, `e.data.handlers` is an empty array. Perhaps I should clarify this with a unit test?